packages/gatsby-recipes/package.json

```diff
@@ -80,36 +59,39 @@
     "single-trailing-newline": "^1.0.0",
     "strip-ansi": "^6.0.0",
     "style-to-object": "^0.3.0",
-    "subscriptions-transport-ws": "^0.9.16",
-    "svg-tag-names": "^2.0.1",
     "unified": "^8.4.2",
     "unist-util-remove": "^2.0.0",
     "unist-util-visit": "^2.0.2",
-    "urql": "^1.9.7",
     "uuid": "3.4.0",
     "ws": "^7.3.0",
     "xstate": "^4.9.1",
-    "yoga-layout-prebuilt": "^1.9.6",
-    "yup": "^0.27.0"
+    "yoga-layout-prebuilt": "^1.9.6"
   },
   "devDependencies": {
     "@babel/cli": "^7.11.6",
+    "@mdx-js/mdx": "^2.0.0-next.4",
+    "@mdx-js/react": "^2.0.0-next.4",
+    "@mdx-js/runtime": "^2.0.0-next.4",
     "@babel/plugin-transform-runtime": "^7.11.5",
     "@rollup/plugin-babel": "^5.1.0",
     "@rollup/plugin-commonjs": "^14.0.0",
     "@rollup/plugin-json": "^4.1.0",
     "@rollup/plugin-node-resolve": "^8.4.0",
+    "@rollup/plugin-replace": "^2.3.3",
     "fetch-mock-jest": "^1.3.0",
     "ink": "next",
     "ink-select-input": "^4.0.0",
     "ink-spinner": "^4.0.0-0",
     "react": "^16.12.0",
+    "react-reconciler": "^0.25.1",
     "rimraf": "^3.0.2",
     "rollup": "^2.23.0",
     "rollup-plugin-auto-external": "^2.0.0",
     "rollup-plugin-internal": "^1.0.0",
+    "subscriptions-transport-ws": "^0.9.16",
     "terminal-link": "^2.0.0",
-    "tmp-promise": "^2.1.0"
+    "tmp-promise": "^2.1.0",
+    "urql": "^1.9.7"
```

Comment:
Hiya :wave: I'm just wondering; I did notice that this dependency on `urql` is getting a little old, so I'm wondering whether anyone on the Recipes team has tried out the newer versions? Also, I'm a little confused about the aim of this PR. I see the problems with MDX above, but a lot of dependencies seem to now selectively either bundled or not bundled at random ✌️

Reply:
I think I put it in there because of `urql`'s `peerDependency` on `react` - if I didn't, then it would resolve to different react version that other things in the recipes cli use, so that caused problems. Please also note that this is CLI react (`ink`) and not web target to make things more confusing.

Essentially I bundled everything that had any connection to `react` here as that was causing problems with `react` version provided by user (but as mention in description - this is special case for the way `npm` install dependencies, we really shouldn't need to do it (and I rather not do this and let package manager properly resolve `node_modules` hierarchy, but that's not the reality unfortunately). So this is workaround on `npm` quirk, rather than problem with the way we, or deps we bundled are setup).

Few of "tickets" that are about similar issues with npm:
 - https://npm.community/t/packages-with-peerdependencies-are-incorrectly-hoisted/4794
 - https://npm.community/t/unmet-peerdependency-but-its-listed-as-a-dependency/5322

Even if those got fixed in recent `npm` version (didn't try), unfortunately our users will also use older versions (like those autoinstalled with Node itself)

As for bundling everything: I did want to limit the scope of bundling to those packages that cause the problems for users (again due to npm quirk, and because packages themselves are problematic). Those that didn't cause (known) problem I did leave as they were - there is some value in not bundling more than needed in node, because there is less duplication as shared packages can be parsed/executed just once (i.e. I explicitly mentioned potential candidates for bundling that I left out not bundled here in https://github.com/gatsbyjs/gatsby/pull/27057#discussion_r499462421 )

Reply:
> I did notice that this dependency on urql is getting a little old

From what I can `latest` of `urql` is `1.11.2`, so this should be covered by that version range - but granted, because we do bundle `urql` now, we will use version that we have specified in lock file

Reply:
That makes sense! The first impression I got was that some packages seemed related to the UI but aren't bundled, which seems a little odd at first.